### ISSUE TITLE
`Generic type 'RawBuilder<O>' requires 1 type argument(s)` in Kysely 0.23

### ISSUE THREAD

**Issue by Qrtn** [RobinBlomberg/kysely-codegen#49]
The generated type `Timestamp` has `RawBuilder` in its union definition, but `RawBuilder` (which now implements `Expression<T>` requires a type parameter `T` corresponding to the type of the expression.

https://github.com/koskimas/kysely/blob/master/recipes/extending-kysely.md#expression

```
export type Timestamp = ColumnType<Date, Date | string | RawBuilder, Date | string | RawBuilder>;
```

**Comment by Mafooch** [comment]
I'm having the same issue in my code as well but I'm also wondering why timestamp is being generated this way to begin with i.e. why is it using RawBuilder. I was expecting something akin to the example given in the docs:
`export type Timestamp = ColumnType<Date, Date | string, Date | string>;`.

**Comment by RobinBlomberg** [comment]
This appears to be related to the new `Expression<T>` interface from kysely@0.23.0.

Thanks for the report! This should be fixed of course, but I'm wondering how to pin the kysely version in a safe manner, right now using `"peerDependencies": { "kysely": ">=0.19.12" }` from earlier advice.

**Comment by koskimas** [comment]
You don't need to explicitly list `RawBuilder` in there. Kysely allows `RawBuilder<T>` if the column type is `T` (or ColumnType with T).

**Comment by koskimas** [comment]
> I'm wondering how to pin the kysely version in a safe manner

We try to only introduce breaking changes in minor versions (major after 1.0). You could use a range up to the next breaking version. But then you need to be updating that each time kysely pushes out a new possibly breaking version.

Things will slow down once we release 1.0 which should be pretty soon.

**Comment by RobinBlomberg** [comment]
> You shouldn't explicitly list `RawBuilder` in there. Kysely allows `RawBuilder<T>` if the column type is `T` (or `ColumnType` with T).

Interesting! I added it because someone reported a bug when it wasn't there, but I will double-check that it can be safely removed.

>Things will slow down once we release 1.0 which should be pretty soon.

Awesome!

**Comment by koskimas** [comment]
I think I was the one who suggested using a peerDependency. That was because I wanted to prevent users having multiple copies of Kysely in a project. That effort seems futile, so we ended up fixing the major problems with having multiple versions 😄

You might consider making Kysely a hard dependency.

**Comment by koskimas** [comment]
> Interesting! I added it because someone reported a bug when it wasn't there, but I will double-check that it can be safely removed.

That seems like a bug in Kysely and should be fixed there.

**Comment by RobinBlomberg** [comment]
Fixed in `kysely-codegen@0.9.0`!
